### PR TITLE
Use prettyprinter-compat-ansi-wl-pprint as an implementation

### LINKED ISSUE #26: This package is deprecated. Use prettyprinter.
https://hackage.haskell.org/package/prettyprinter

See #18 

`ansi-wl-pprint` may or may not receive (timely) updates.

### PR DESCRIPTION
**Pull request by phadej**
Resolves #26, as with this patch you'll end up using `prettyprinter`.

---

Before:

Screenshot from 2023-05-16 18-01-29

After:

Screenshot from 2023-05-16 18-00-14

The green text is done with

```haskell
    -- ...but going via a string will only preserve formatting information information on Unix
    putStr $ show $ green (text "I will be green on Unix but uncolored on Windows") <> linebreak
```

and that doesn't work with  `Prettyprinter`'s `Doc` (which doesn't use annotations in `Show` instance).

Otherwise compat shim seems to be quite complete, as none of imports needed to be removed.

This will finally make `ansi-wl-pprint` obsolete.

### CODE REVIEW

**Comment by phadej** [comment]
@RyanGlScott, do you have an opinion on this?

File: ansi-wl-pprint.cabal [hunk 1]
```diff
@@ -1,7 +1,6 @@
 cabal-version:       >= 1.10
 name:                ansi-wl-pprint
-version:             0.6.9
-x-revision:          4
+version:             1.0.2
```
Comment (RyanGlScott) [inline ansi-wl-pprint.cabal:3]: Why pick version `1.0.2` instead of, say, `1.0.0`?

File: Text/PrettyPrint/ANSI/Leijen.hs [hunk 2]
```diff
@@ -158,74 +160,5 @@ module Text.PrettyPrint.ANSI.Leijen (
...
-
-#if __GLASGOW_HASKELL__ >= 710
-import Data.Monoid ((<>))
-#elif __GLASGOW_HASKELL__ >= 704
-import Data.Monoid (Monoid, mappend, mconcat, mempty, (<>))
-#else
-import Data.Monoid (Monoid, mappend, mconcat, mempty)
-#endif
-
--- $DocumentAlgebra
```
Comment (RyanGlScott) [inline Text/PrettyPrint/ANSI/Leijen.hs:171]: Did you mean to delete these Haddocks? `$DocumentAlgebra` is referenced in the export list.

File: Changelog.md [hunk 3]
```diff
@@ -1,8 +1,19 @@
+## 1.0.2
+
+- Change implementation to the `prettyprinter` package,
+  using `prettyprinter-compat-ansi-wl-pprint` compatibility package.
+
+  Migrate to `prettyprinter`.
+
+  Note: Going through String won't preserve formatting anymore,
+  as `prettyprinter`s `Show Doc` instance is annotation invariant.
```
Comment (RyanGlScott) [inline Changelog.md]: Is there a way to migrate code affected by this change in behavior?

File: Changelog.md [hunk 4]
```diff
@@ -1,8 +1,19 @@
+## 1.0.2
+
+- Change implementation to the `prettyprinter` package,
+  using `prettyprinter-compat-ansi-wl-pprint` compatibility package.
```
Comment (RyanGlScott) [inline Changelog.md]: English-language nit picks:

```suggestion
- Change the implementation to the `prettyprinter` package,
  using the `prettyprinter-compat-ansi-wl-pprint` compatibility package.
```

File: Changelog.md [hunk 5]
```diff
@@ -1,8 +1,19 @@
...
+
+  Migrate to `prettyprinter`.
+
+  Note: Going through String won't preserve formatting anymore,
+  as `prettyprinter`s `Show Doc` instance is annotation invariant.
+
+- `Text.PrettyPrint.ANSI.Leijen.Internal` module is gone.
```
Comment (RyanGlScott) [inline Changelog.md]:
```suggestion
- The `Text.PrettyPrint.ANSI.Leijen.Internal` module is gone.
```

**Review by RyanGlScott — COMMENTED** [review]
This looks great!

File: ansi-wl-pprint.cabal (hunk 1 above)
Comment (phadej) [reply ansi-wl-pprint.cabal:3]: It's the same version as https://hackage.haskell.org/package/prettyprinter-compat-ansi-wl-pprint

File: Changelog.md (hunk 3 above)
Comment (phadej) [reply Changelog.md]: In fact yes. I'll add a note. TL;DR https://hackage.haskell.org/package/prettyprinter-ansi-terminal-1.1.3/docs/Prettyprinter-Render-Terminal.html#v:renderLazy

File: Text/PrettyPrint/ANSI/Leijen.hs (hunk 2 above)
Comment (phadej) [reply Text/PrettyPrint/ANSI/Leijen.hs:171]: undone

File: Changelog.md (hunk 3 above)
Comment (phadej) [reply Changelog.md]: Added

File: Changelog.md [hunk 6]
```diff
@@ -1,8 +1,22 @@
...
+- Change the implementation to the `prettyprinter` package,
+  using the `prettyprinter-compat-ansi-wl-pprint` compatibility package.
+
+  Migrate to `prettyprinter`.
+
+  Note: Using `show` won't preserve formatting anymore,
+  as `prettyprinter`s `Show Doc` instance is annotation invariant.
+  You can convert to `String` using
+  `Data.Text.Lazy.unpack . Prettyprinter.Render.Terminal.renderLazy . Prettyprinter.layoutPretty Prettyprinter.defaultLayoutOptions`,
+  from `text`, `prettyprinter-ansi-terminal`, `prettyprinter` packages respectively.
```
Comment (RyanGlScott) [inline Changelog.md]:
```suggestion
  from the `text`, `prettyprinter-ansi-terminal`, `prettyprinter` packages respectively.
```

**Review by RyanGlScott — APPROVED** [review]

**Comment by georgefst** [comment]
Note that there are other incompatibilities: https://github.com/quchen/prettyprinter/issues/246. These ideally ought to have been mentioned in the release notes, but at least this package was given a major version bump when switching over.

**Pull request merged**